test/scale/scale_tester.go

```diff
@@ -178,94 +186,115 @@ scaleInTest:
 	tStorageClass, storageCleanup := t.Pod.CreateStorageClass(client, namespace, t.CSIDriver, t.StorageClassParameters)
 	defer storageCleanup()
+	statefulSet, cleanupStatefulSet := t.Pod.SetupStatefulset(client, namespace, t.CSIDriver, schedulerName, t.Replicas, &tStorageClass, map[string]string{"group": "azuredisk-scale-tester"})
+	defer cleanupStatefulSet(45 * time.Minute)
+
+	statefulSet.CreateWithoutWaiting()
+	ticker := time.NewTicker(time.Minute)
+	numberOfRunningPods1 := podutil.WaitForPodsWithMatchingLabelToRun(*scaleTestTimeout, ticker, totalNumberOfPods, client, namespace)
+	scaleOutCompleted := time.Since(startScaleOut).Minutes()
 
-	for i := 0; i < t.PodCount; i++ {
-		ss, cleanup := t.Pod.SetupStatefulset(client, namespace, t.CSIDriver, schedulerName, t.Replicas, &tStorageClass, nil)
-		statefulSets = append(statefulSets, ss)
-		defer cleanup(45 * time.Minute)
+	// cordon the first half nodes where the running pods sit on
+	for _, node := range nodes[:midNode] {
+		node := node
+		framework.Logf("Cordoning %s.", node)
+		testPod.SetNodeUnschedulable(node, true)
 	}
 
-	ginkgo.By("Completed statefulset setup")
+	// uncordon the second nodes and start to create replica attachments
+	for _, node := range nodes[midNode:] {
+		node := node
+		framework.Logf("Uncordoning %s.", node)
+		testPod.SetNodeUnschedulable(node, false)
+	}
 
-	ginkgo.By("Initiating statefulset deployment")
-	var wg sync.WaitGroup
-	start := time.Now()
-	for i := range statefulSets {
-		wg.Add(1)
-		go func(statefulset *resources.TestStatefulset) {
-			defer wg.Done()
-			defer ginkgo.GinkgoRecover()
+	roleReq, err := azureutils.CreateLabelRequirements(consts.RoleLabel, selection.Equals, string(azdiskv1beta2.ReplicaRole))
+	framework.ExpectNoError(err)
+	labelSelector := labels.NewSelector().Add(*roleReq)
 
-			statefulset.Create()
-		}(statefulSets[i])
-	}
+	ticker.Reset(time.Minute)
+	tickerCount := 0
+	timeout := time.After(time.Duration(*scaleTestTimeout) * time.Minute)
+	startCreateReplicaAtts := time.Now()
+	numberOfAttachedReplicaAtt := 0
 
-	ginkgo.By("Waiting for the pod for all the statefulsets to succeed")
-	wg.Wait()
-	framework.Logf("Time taken to complete deployment of statefulsets : %d", time.Since(start).Milliseconds())
-
-	ginkgo.By("All pods for the statefulsets are ready")
-	ginkgo.By("Initiating scaling down all the statefulsets")
-
-	for i := range statefulSets {
-		newScale := &scale.Scale{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      statefulSets[i].Statefulset.Name,
-				Namespace: statefulSets[i].Namespace.Name,
-			},
-			Spec: scale.ScaleSpec{
-				Replicas: int32(0),
-			},
-		}
+	maxShares, _ := azureutils.GetMaxSharesAndMaxMountReplicaCount(t.StorageClassParameters, false)
+	totalNumberOfReplicaAtts := totalNumberOfPods * (maxShares - 1)
 
-		go func(ss *resources.TestStatefulset) {
-			defer wg.Done()
-			defer ginkgo.GinkgoRecover()
+waitForReplicaAtts:
+	for {
+		select {
+		case <-timeout:
+			break waitForReplicaAtts
```

Comment:
Let's add a log line to print the number of replicas when the test times out to give us some idea of how much progress it was able to make

Reply:
Thanks for the reminder. I also add that if `numberOfAttachedReplicaAtts != totalNumberOfReplicaAtts` will fail the test.